source/common/filesystem/win32/filesystem_impl.cc

```diff
@@ -119,7 +120,12 @@ bool InstanceImplWin32::directoryExists(const std::string& path) {
 }
 
 ssize_t InstanceImplWin32::fileSize(const std::string& path) {
-  auto fd = CreateFileA(path.c_str(), GENERIC_READ, FILE_SHARE_READ, 0, OPEN_EXISTING, 0, NULL);
+  // In ssl_integration_test::SslKeyLogTest cases, a temporary file is created, and then when
+  // we check whether some logs are printed into the file, CreateFileA will fail and report the
```

Comment:
Actually, if FILE_SHARE__WRITE is not set, it will always report the error: "The process cannot access the file because it is being used by another process". I just refer to the function fileReadToEnd which also includes the FILE_SHARE_WRITE although I do not why it is needed for read.

Reply:
@ggreenway , in other tests, it seems that they do not use the temp file. I am not sure whether it is the cause.